### PR TITLE
Store correct block ID when switching forks

### PR DESCRIPTION
**Pull request by abitmore**
Related to #831 .

### CODE REVIEW

**Comment by pmconrad** [comment]
Looks good. @jmjatlanta are you working on a test case?

**Comment by jmjatlanta** [comment]
I was not working on a test case, but I will.

File: tests/tests/block_tests.cpp [hunk 1]
```diff
@@ -269,25 +276,36 @@ BOOST_AUTO_TEST_CASE( fork_blocks )
...
+         BOOST_CHECK_EQUAL(b.block_num(), 14u);
          GRAPHENE_CHECK_THROW(PUSH_BLOCK( db1, b ), fc::exception);
+         // Issue #938 make sure db is in a good state
+         fc::optional<signed_block> previous_block = db1.fetch_block_by_number(1);
+         BOOST_CHECK ( previous_block.valid() );
+         for( uint32_t curr_block_num = 2; curr_block_num <= 13; ++curr_block_num )
+         {
+            fc::optional<signed_block> curr_block = db1.fetch_block_by_number( curr_block_num );
+            BOOST_CHECK( curr_block.valid() );
+            BOOST_CHECK_EQUAL( curr_block->previous, previous_block->id() );
```
Comment (jmjatlanta) [inline tests/tests/block_tests.cpp]: This test should fail with the old code, but is not.

Comment (abitmore) [reply tests/tests/block_tests.cpp]: `fetch_block_by_number` here will read data from `_fork_db`, thus will pass the check.

I think we can add more tests in the end: close db, reopen db, run this check again.

Comment (jmjatlanta) [reply tests/tests/block_tests.cpp]: Unfortunately, the original problem still does not show up. I haven't dug too deeply, but I'm guessing the reindexing fixes things when the database reopens.

Comment (abitmore) [reply tests/tests/block_tests.cpp]: Oh, reopen at this point doesn't work. Try generate 30 more blocks to push the 2 blocks out of `fork_db`.

Comment (abitmore) [reply tests/tests/block_tests.cpp]: Oh, seems due to empty blocks. I'm trying to update the test case.

Comment (abitmore) [reply tests/tests/block_tests.cpp]: My comments earlier are incorrect.

Actually `PUSH_BLOCK( db1, good_block );` made all blocks stored in `db1` correct. I'm trying to reproduce.

**Comment by abitmore** [comment]
@pmconrad @jmjatlanta @oxarbitrage test case fixed, please review again.

**Pull request merged**